### ISSUE TITLE
Reflect the current mining mode in the status header

### ISSUE THREAD

**Issue by ivpavici** [0xSpaceShard/starknet-rivet#102]
Currently, the status bar only has hardcoded "AUTO" string for block mining.

We should change this based on the mining mode, the other option being: https://0xspaceshard.github.io/starknet-devnet-rs/docs/next/blocks#creating-blocks-on-demand  

Additionally (but this can be a separate issue) - the block number is updated only after clicking on BLOCKS

![image](https://github.com/user-attachments/assets/0fbafd89-4b72-4051-b905-211f9b2a5457)
